plugins/pulp_docker/plugins/importers/sync.py

```diff
@@ -290,25 +294,29 @@ def _process_manifest(self, manifest, digest, available_blobs, tag=None):
         :param available_blobs: set of current available blobs accumulated dusring sync
         :type available_blobs: set
 
-        :return: An initialized Manifest object
-        :rtype: pulp_docker.plugins.models.Manifest
-
+        :return: a boolean which indicates if the Manifest has foreign layers
+        :rtype: bool
         """
 
         # Save the manifest to the working directory
         with open(os.path.join(self.get_working_dir(), digest), 'w') as manifest_file:
             manifest_file.write(manifest)
         manifest = models.Manifest.from_json(manifest, digest)
         self.parent.available_manifests.append(manifest)
+        has_foreign_layer = False
         for layer in manifest.fs_layers:
-            available_blobs.add(layer.blob_sum)
+            if layer.layer_type and layer.layer_type == constants.FOREIGN_LAYER:
```

Comment:
I'm still not sure why we need to test ``if layer.layer_type`` here, because if it's any value that evaluates to ``False``, that would not be equal to ``constants.FOREIGN_LAYER``. So I think this can be simplified to ``if layer.layer_type == constants.FOREIGN_LAYER:`` That said, the end behavior is still correct.